src/commands/trace/trace.ts

```diff
@@ -0,0 +1,121 @@
+import chalk from 'chalk'
+import {spawnSync} from 'child_process'
+import {Command} from 'clipanion'
+import crypto from 'crypto'
+import os from 'os'
+import {performance} from 'perf_hooks'
+import {parseTags} from '../../helpers/tags'
+import {apiConstructor} from './api'
+import {APIHelper} from './interfaces'
+
+export class TraceCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Trace a command with a custom span and report it to Datadog.',
+    details: `
+            This command wraps another command, which it will launch, and report a custom span to Datadog.
+            See README for details.
+        `,
+    examples: [
+      ['Trace a command and report to Datadog', 'datadog-ci trace echo "Hello World"'],
+      [
+        'Trace a command and report to the datadoghq.eu site',
+        'DATADOG_SITE=datadoghq.eu datadog-ci trace echo "Hello World"',
+      ],
+    ],
+  })
+
+  private command?: string[]
+
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY || process.env.DD_API_KEY,
+    envVarTags: process.env.DD_TAGS,
+  }
+
+  public async execute() {
+    if (!this.command || !this.command.length) {
+      this.context.stderr.write('Missing command to run\n')
+
+      return 1
+    }
+
+    const [command, ...args] = this.command
+    const id = crypto.randomBytes(5).toString('hex')
+    const t0 = performance.now()
+    const spawnResult = spawnSync(command, args, {env: {...process.env, DD_CUSTOM_PARENT_ID: id}, stdio: 'inherit'})
+    const t1 = performance.now()
+    const duration = t1 - t0
+    const exitCode = spawnResult.status ?? this.signalToNumber(spawnResult.signal!) ?? 127
+    const api = this.getApiHelper()
+    const [provider, data] = this.getData()
+    await api.reportCustomSpan(
+      {
+        data,
+        duration,
+        id,
+        is_error: exitCode !== 0,
+        parent_id: process.env.DD_CUSTOM_PARENT_ID,
+        tags: this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {},
+      },
+      provider
+    )
+
+    return exitCode
+  }
+
+  public getData(): [string, Record<string, string>] {
+    if (process.env.CIRCLECI === 'true') {
```

Comment:
I think you can get away with 
```suggestion
    if (process.env.CIRCLECI) {
```